core/src/main/java/org/bitcoinj/core/PeerGroup.java

```diff
@@ -518,6 +518,9 @@ public void go() {
 
             boolean doDiscovery = false;
             long now = Utils.currentTimeMillis();
+            if (countConnectedAndPendingPeers() >= getMaxConnections()) {
```

Comment:
why not do this "early-out" together with the other one, at the beginning of the method:

````
 if (!vRunning || countConnectedAndPendingPeers() >= getMaxConnections())) {return;}
````